### PR TITLE
[MM-19267] Add GetPluginStatuses plugin API method and OnPluginStatusesChanged hook

### PR DESCRIPTION
**Pull request by KushalBeniwal**
#### Summary

Add GetPluginStatuses for plugin API.

Pull Request to fix [MM-19267](https://mattermost.atlassian.net/browse/MM-19267).

#### To-Do

- [x] Add GetPluginStatuses plugin API method
- [ ] Add OnPluginStatusesChanged hook

#### Ticket Link

Fixes [#13945](https://github.com/mattermost/mattermost-server/issues/13945)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by KushalBeniwal** [comment]
The issue required adding `GetPluginStatuses` and `OnPluginStatusesChanged`, so I thought of making two different pull requests to keep the PRs short and easy to review.
Though this is my thinking and can be wrong in this scenario, so please do let me know if that's the case and I'll make the required changes in this PR!

Also, a question:
Initially the hook was supposed to be added for `App.notifyPluginStatusesChanged` ([here](https://github.com/mattermost/mattermost-server/blob/5e69c6b02f604ce1aecb4c22cc58f9412cc685a2/app/plugin_statuses.go#L95)) but from then there has been two major changes in the file changing the receiver from App to Server ([here](https://github.com/mattermost/mattermost-server/commit/5ea06e51d08f6f960029c981a48986681968f34f)) and then Server to Channels ([here](https://github.com/mattermost/mattermost-server/commit/25257d6ef660b3f29be4d7294582964472cab6ae)). Will these design changes also require the issue to change in any aspect? 
PS just wanted to know if the requirements changed over the span of last two years!

**Review by hanzei — APPROVED** [review]
LGTM, thanks for the contribution :+1:

File: app/plugin_api_tests/test_get_plugin_statuses_plugin/main.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+func (p *MyPlugin) MessageWillBePosted(_ *plugin.Context, _ *model.Post) (*model.Post, string) {
+	statuses, err := p.API.GetPluginStatuses()
+	if err != nil {
+		return nil, err.Error()
+	}
+
+	if len(statuses) != 1 {
+		return nil, "Returned invalid number of plugin statuses"
+	}
```
Comment (mickmister) [inline app/plugin_api_tests/test_get_plugin_statuses_plugin/main.go:32]: Maybe assert that the correct plugin is running in the response?

**Review by mickmister — APPROVED** [review]
Thanks @KushalBeniwal! LGTM :+1:

**Comment by mickmister** [comment]
> The issue required adding `GetPluginStatuses` and `OnPluginStatusesChanged`, so I thought of making two different pull requests to keep the PRs short and easy to review. Though this is my thinking and can be wrong in this scenario, so please do let me know if that's the case and I'll make the required changes in this PR!

If these features are coupled on a UX level (which it seems they are), then they should be put into one PR. The changes should be fairly easy to review in general, since the actual changes are pretty minimal. What do you think @KushalBeniwal @hanzei?

**Pull request closed without merging**